numpy/core/_methods.py

```diff
@@ -165,7 +165,7 @@ def _mean(a, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
     is_float16_result = False
 
     rcount = _count_reduce_items(arr, axis, keepdims=keepdims, where=where)
-    if rcount == 0 if where is True else umr_any(rcount == 0):
+    if (where is True and rcount == 0) or (where is not True and _any(rcount == 0)):
```

Comment:
So, after puzzling longer... The error I made is the missing `axis=None` here. It seems that unlike our top-level aliases, `ufunc.reduce` uses `axis=0` by default. And I frankly would not have known that without stumbling over it even otherwise...

It was silly to delete it, but just as bad that none of these have tests that fail here. Could you use the `axis=None` solution (could pass by position if you like), and add a simple test?

Reply:
@seberg Sure! I've just pushed changes reverting to `umr_any` and adding test case